src/client/Microsoft.Identity.Client/OAuth2/TokenClient.cs

```diff
@@ -130,7 +133,15 @@ private void AddBodyParamsAndHeaders(IDictionary<string, string> additionalBodyP
             }
 
             _oAuth2Client.AddBodyParameter(OAuth2Parameter.Scope, scopes);
+
+#if SUPPORT_KERBEROS
+            // Add Kerberos Ticket claims if there's valid service principal name in Configuration.
+            // Kerberos Ticket claim is only allowed at token request due to security issue.
+            // It should not be included for authorize request.
+            KerberosSupplementalTicketManager.AddKerberosTicketClaim(_oAuth2Client, _requestParams);
+#else
```

Comment:
I wonder if we should merge these...

Reply:
Yes. Kerberos Ticket claim should be merged into existing claims.